FWCore/Framework/src/InputSource.cc

```diff
@@ -84,12 +83,9 @@ namespace edm {
     }
     if (maxSecondsUntilRampdown_ > 0) {
       processingStart_ = std::chrono::steady_clock::now();
-  }
-
-    // Secondary input sources currently do not have a product registry.
-    if(primary_) {
-      assert(desc.productRegistry_ != 0);
     }
+
+    assert(desc.productRegistry_ != 0);
```

Comment:
Would it make more sense now for the ```InputSourceDescription``` to enforce this itself since it is only used with a primary source?